### ISSUE TITLE
vimbtool.php -a mailbox.cli-get-sizes not working

### ISSUE THREAD

**Issue by itabv** [opensolutions/ViMbAdmin#301]
I have the latest version 3.3.0 installed. PHP 8.2. Mailbox sizing function is not working
```

root@mailsrv:~# /opt/vimbadmin/bin/vimbtool.php -a mailbox.cli-get-sizes
PHP Deprecated:  Creation of dynamic property Bootstrap::$frontController is deprecated in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Resource/Frontcontroller.php on line 161

Deprecated: Creation of dynamic property Bootstrap::$frontController is deprecated in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Resource/Frontcontroller.php on line 161
PHP Warning:  ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 211

Warning: ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 211
PHP Warning:  ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226

Warning: ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226
PHP Warning:  ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226

Warning: ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226
PHP Warning:  ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226

Warning: ini_set(): Session ini settings cannot be changed after headers have already been sent in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php on line 226
Zend_Session_Exception Object
(
    [message:protected] => Session must be started before any output has been sent to the browser; output started in /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Resource/Frontcontroller.php/161
    [string:Exception:private] => 
    [code:protected] => 0
    [file:protected] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session.php
    [line:protected] => 466
    [trace:Exception:private] => Array
        (
            [0] => Array
                (
                    [file] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Session/Namespace.php
                    [line] => 143
                    [function] => start
                    [class] => Zend_Session
                    [type] => ::
                )

            [1] => Array
                (
                    [file] => /opt/vimbadmin/library/OSS/Resource/Namespace.php
                    [line] => 81
                    [function] => __construct
                    [class] => Zend_Session_Namespace
                    [type] => ->
                )

            [2] => Array
                (
                    [file] => /opt/vimbadmin/library/OSS/Resource/Namespace.php
                    [line] => 63
                    [function] => getSession
                    [class] => OSS_Resource_Namespace
                    [type] => ->
                )

            [3] => Array
                (
                    [file] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Bootstrap/BootstrapAbstract.php
                    [line] => 695
                    [function] => init
                    [class] => OSS_Resource_Namespace
                    [type] => ->
                )

            [4] => Array
                (
                    [file] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Bootstrap/BootstrapAbstract.php
                    [line] => 638
                    [function] => _executeResource
                    [class] => Zend_Application_Bootstrap_BootstrapAbstract
                    [type] => ->
                )

            [5] => Array
                (
                    [file] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Bootstrap/BootstrapAbstract.php
                    [line] => 598
                    [function] => _bootstrap
                    [class] => Zend_Application_Bootstrap_BootstrapAbstract
                    [type] => ->
                )

            [6] => Array
                (
                    [file] => /opt/vimbadmin/vendor/shardj/zf1-future/library/Zend/Application.php
                    [line] => 389
                    [function] => bootstrap
                    [class] => Zend_Application_Bootstrap_BootstrapAbstract
                    [type] => ->
                )

            [7] => Array
                (
                    [file] => /opt/vimbadmin/bin/vimbtool.php
                    [line] => 45
                    [function] => bootstrap
                    [class] => Zend_Application
                    [type] => ->
                )

        )

    [previous:Exception:private] => 
    [_previous:Zend_Exception:private] => 
)
```

**Comment by nine** [comment]
ViMbAdmin supports PHP 8.1. Please check the docs and use the right PHP version:

https://github.com/opensolutions/ViMbAdmin/releases/tag/3.3.0

**Comment by itabv** [comment]
PHP8.1 solved my problem. I can not find where the maximum version of PHP is said. I see only minimal

**Comment by sm4rty-github** [comment]
Workaround:
Edit: vimbadmin/vendor/shardj/zf1-future/library/Zend/Application/Bootstrap/BootstrapAbstract.php

change: 

```
abstract class Zend_Application_Bootstrap_BootstrapAbstract 
    implements Zend_Application_Bootstrap_Bootstrapper,
               Zend_Application_Bootstrap_ResourceBootstrapper
```

to:

```
abstract class Zend_Application_Bootstrap_BootstrapAbstract extends \stdClass.
    implements Zend_Application_Bootstrap_Bootstrapper,
               Zend_Application_Bootstrap_ResourceBootstrapper
```

Edit:
vimbadmin/vendor/smarty/smarty/libs/sysplugins/smarty_internal_extension_handler.php

change:

`class Smarty_Internal_Extension_Handler`

to:

`class Smarty_Internal_Extension_Handler extends \stdClass`

Now its wörking.

**Comment by nine** [comment]
Please consider to make a Github pull request with this fix. Thanks

**Comment by sm4rty-github** [comment]
![grafik](https://user-images.githubusercontent.com/121514749/218265389-9ee54aa4-9170-4d87-ab6b-78419608a529.png)

Dont know how to do.

**Comment by barryo** [comment]
Should be fixed now via composer update for php >= 8.0. Release coming shortly.
